pkg/debuginfo/debuginfo.go

```diff
@@ -52,8 +51,8 @@ func New(logger log.Logger, client Client) *DebugInfo {
 		logger: logger,
 		client: client,
 		existsCache: cache.New(
-			cache.WithMaximumSize(128),                // Arbitrary cache size.
-			cache.WithExpireAfterWrite(2*time.Minute), // Arbitrary period.
+			cache.WithMaximumSize(512), // Arbitrary cache size.
+			cache.WithPolicy("lru"),
```

Comment:
We need to keep a time-based eviction strategy. 
As Frederic mentioned, things can change on the Parca server side. Invalid or corrupted information can be deleted on the server-side. 
The single source of truth should be the server. However, we don't want to overwhelm the server with the "exists" queries that's why we added the cache and set an expiration date so that symbolization would eventually be consistent.